packages/atomic-layout/src/hooks/useResponsiveProps.ts

```diff
@@ -3,41 +3,64 @@ import {
   Numeric,
   Layout,
   createMediaQuery,
+  ParsedProp,
   parsePropName,
 } from '@atomic-layout/core'
 import useBreakpointChange from './useBreakpointChange'
 
+type MatcherFunction = (parsedProp: ParsedProp) => boolean
+
+/**
+ * Default responsive props matcher.
+ * Creates a media query based on the given prop's breakpoint
+ * and uses native "window.matchMedia" to assert the match.
+ */
+const defaultMatcher: MatcherFunction = (parsedProp) => {
+  const { breakpoint, behavior } = parsedProp
+  const mediaQuery = createMediaQuery(
+    Layout.breakpoints[breakpoint.name],
+    behavior,
+  )
+
+  return matchMedia(mediaQuery).matches
+}
+
+/**
+ * Filters given responsive props against the browser state.
+ * Accepts an optional matcher function to operate on a server.
+ */
+const filterProps = <R>(
+  props: Record<string, any>,
+  matcher: MatcherFunction = defaultMatcher,
+) => {
+  return Object.keys(props)
+    .map(parsePropName)
+    .filter(matcher)
+    .reduce<R>(
+      (acc, { originPropName, purePropName }) => ({
+        ...acc,
+        [purePropName]: props[originPropName],
+      }),
+      {} as R,
+    )
+}
+
 /**
  * Accepts an object of responsive props and returns
  * an object of props relative to the current viewport.
  */
 const useResponsiveProps = <ResponsiveProps extends Record<string, Numeric>>(
   responsiveProps: ResponsiveProps,
 ): Partial<ResponsiveProps> => {
-  const [props, setProps] = useState<ResponsiveProps>()
+  const [props, setProps] = useState<ResponsiveProps>(
+    filterProps(responsiveProps, ({ breakpoint }) => {
+      return breakpoint.isDefault && typeof window === 'undefined'
```

Comment:
you can omit the return, more concise 💯

Reply:
Call it a nitpick, but I like explicit returns. Makes less things to change if you wish to log something before return, and the keyword is a great eye-catcher to see when the function ends.